src/bridge/cockpithttpstream.c

```diff
@@ -215,6 +215,11 @@ typedef struct _CockpitHttpStream {
 
   /* The request */
   GList *request;
+  gint64 body_length;
+
+  /* Stores number of bytes in the request waiting to be sent */
+  /* Only valid when body_length is set (otherwise the request is sent at once) */
+  gulong request_buffer_size;
```

Comment:
I don't understand why we do this rather than just sending immediately, once the header is sent.  `cockpit_stream_write()` is also doing its own buffering.

Also: the buffering in the stream seems to already be throttling-enabled.  It might be nice to write a test to make sure this is functioning properly for large POST requests with e.g. a fake webserver which lets the connection fill up, and making sure that the connection exerts back-pressure properly.  Look at cockpitflow for more information about how that should work.

Reply:
Good point. Simply sending the received data via `cockpit_stream_write` on receive seems to be very slow when sending small pieces of data multiple times (like the "HelloWorld" string in the tests), though. I'll have a look at what exactly is happening there.